Unstructured-IO/unstructured-api
# chore: build with registry as cache

* updates docker-build.sh to use the image registry as a cache
* updates make commands to no longer explicitly set platform (but optionally can be assigned via env var)


## Testing

Previous CI docker-test run time: [32m](https://github.com/Unstructured-IO/unstructured-api/actions/runs/4616209135/jobs/8160967137) 
New CI docker-test run time: [7m20s](https://github.com/Unstructured-IO/unstructured-api/actions/runs/4623792027/jobs/8177994053?pr=71) 

- Run `make docker-build` and note that most of steps will say CACHED.
- Run `make docker-build` on amd vs arm and use `docker inspect image unstructured-api:dev | grep Architecture` and note that the respective image for your platform was built

Also, I triggered a publish flow to validate that this still works as expected. Here is that run: https://github.com/Unstructured-IO/unstructured-api/actions/runs/4631349496
